tee-worker/service/src/parentchain_handler.rs

```diff
@@ -142,6 +146,49 @@ where
 				return Ok(until_synced_header)
 			}
 
+			let mut events: Vec<Events> = vec![];
+			for block in &block_chunk_to_sync {
+				let block_events = self.parentchain_api.events(Some(block.block.hash()))?;
+				events.push(block_events);
+			}
+			block_chunk_to_sync
+				.iter_mut()
+				.enumerate()
+				.for_each(|(block_index, signed_block)| {
+					let mut extrinsics: Vec<OpaqueExtrinsic> = vec![];
+					let block_events = events.get(block_index).unwrap();
+					for (i, xt) in signed_block.block.extrinsics.iter().enumerate() {
+						// Check if the tx was successful
+						let success = block_events
+							.iter()
+							.filter(|event| {
+								if let Ok(event) = event {
+									event.pallet_name() == "System"
+										&& event.variant_name() == "ExtrinsicSuccess"
+										&& event.phase() == Phase::ApplyExtrinsic(i as u32)
+								} else {
+									false
+								}
+							})
+							.count() > 0;
```

Comment:
Additional question, is `count()` always 1 if the extrinsic succeeds? Any cases where count() can be > 1?